tinygrad/codegen/linearizer.py

```diff
@@ -90,7 +90,7 @@ def global_load(self, i:int, idxs:List[Node], acc=None, barrier:Optional[UOp]=No
     invalid_value = 0 if dtypes.is_int(buf.dtype) else 0.0
     for idx, valid, rep_idx in zip(e_idxs, e_valids, iter_idxs(expand_vars)):
       this_const, idx, valid = (invalid_value, NumNode(0), NumNode(1)) if valid.max == 0 else (const, idx, valid)
-      key = f"{acc}{localtype}{this_const if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
+      key = f"{acc}{localtype}{'CONST'+str(this_const) if this_const is not None and acc is None else (buf.idx if isinstance(buf, MemBuffer) else cast(LocalBuffer, buf).name)}{idx.render()}{valid.render()}"  # noqa: E501
```

Comment:
cache keys can have conflicts between ConstBuffers and MemBuffers, see:
https://github.com/tinygrad/tinygrad/pull/3401#issuecomment-1944684081

we didnt have this problem before because MemBuffers are indexed with ints and [ConstBuffers where always float](https://github.com/tinygrad/tinygrad/pull/3401/files#diff-4fe1256664cc5870d2d44de21d5f463a512f81729a13f5309e272c8f7793a039L96)